src/java.base/share/classes/sun/security/ssl/JsseJce.java

```diff
@@ -162,10 +188,38 @@ private static class EcAvailability {
         // Is EC crypto available?
         private static final boolean isAvailable;
 
+        /**
+         * Checks if a particular signature algorithm is available.
+         *
+         * @param algorithm The algorithm we will attempt to instantiate to check if it is available
+         * @return true if the signature algorithm is found, false otherwise
+         */
+        private static boolean isSignatureAlgorithmAvailable(String algorithm) {
+            try {
+                // Attempt to create a Cipher instance with the specified algorithm.
+                Signature.getInstance(algorithm);
+                return true;
+            } catch (NoSuchAlgorithmException e) {
+                return false;
+            }
+        }
+
         static {
             boolean mediator = true;
             try {
-                Signature.getInstance(SIGNATURE_ECDSA);
+                // When running in FIPS mode, the signature "SHA1withECDSA" is not
+                // available by default. In this scenario we should still set EC
+                // availability to true since other algorithms in the ECDSA signature
+                // family are available for use in various ECDSA TLS ciphers. All
+                // FIPS solutions are expected to have an algorithm such as
+                // "SHA512withECDSA", "SHA384withECDSA", "SHA256withECDSA", or
+                // "SHA224withECDSA" available so we will also check for these algorithms.
```

Comment:
The commit message should also be updated to spell "FIPS" consistently and make it clearer (like this comment does) that we don't only check for `SHA1withECDSA` and `SHA512withECDSA`.

Reply:
FIPS is now consistently upper cased. 

The wording for the commit message was also adjusted to list out various ECDSA algorithms.

The PR message was copied from the commit message to force them to match.

Reply:
Please fix the first paragraph of the commit message: finish the first sentence with a period and start the second with a capital.
```
In FIPS mode, certain cryptographic algorithms are typically disabled.
This includes the algorithm `SHA1withECDSA` since `SHA1` is no longer
recommended for general use.
```

Reply:
This sentence was updated as per above.